### PR TITLE
Drop support for 2021.3 (closes #785)

### LINKED ISSUE #785: 'panel(...): DialogPanel' is deprecated; use com.intellij.ui.dsl.builder.panel from Kotlin UI DSL Version 2
```
w: /home/plipski/git-machete-intellij-plugin/frontendActions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/DeleteBranchOnSlideOutSuggestionDialog.kt: (38, 38): 'panel(vararg LCFlags, String? = ..., LayoutBuilder.() -> Unit): DialogPanel' is deprecated. Use com.intellij.ui.dsl.builder.panel from Kotlin UI DSL Version 2 and kotlin documentations for related classes
w: /home/plipski/git-machete-intellij-plugin/frontendActions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/OverrideForkPointDialog.kt: (40, 38): 'panel(vararg LCFlags, String? = ..., LayoutBuilder.() -> Unit): DialogPanel' is deprecated. Use com.intellij.ui.dsl.builder.panel from Kotlin UI DSL Version 2 and kotlin documentations for related classes
w: /home/plipski/git-machete-intellij-plugin/frontendActions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/SlideInDialog.kt: (44, 38): 'panel(vararg LCFlags, String? = ..., LayoutBuilder.() -> Unit): DialogPanel' is deprecated. Use com.intellij.ui.dsl.builder.panel from Kotlin UI DSL Version 2 and kotlin documentations for related classes
```

### PR DESCRIPTION
**Pull request by mkondratek**
Sadly we need this one to introduce slide in autocompletion (#958).
As discussed... let's wait until 2022.2 is released.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ powerMock = "2.0.9"
 aspectj-postCompileWeaving = "io.freefair.gradle:aspectj-plugin:6.4.3.1"
 checkerFramework = "org.checkerframework:checkerframework-gradle-plugin:0.6.13"
 jetbrains-intellij = "org.jetbrains.intellij.plugins:gradle-intellij-plugin:1.7.0"
-jetbrains-kotlin = "org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.0"
+jetbrains-kotlin = "org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.20"
```
Comment (PawelLipski) [inline gradle/libs.versions.toml:11]: To be unified using `versionRef` on #1004

Comment (PawelLipski) [inline gradle/libs.versions.toml:11]: To be unified using `versionRef` on #1004

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/BuildUtils.kt [hunk 2]
```diff
@@ -16,8 +16,7 @@ object BuildUtils {
     project.apply(plugin = "org.jetbrains.kotlin.jvm")
 
     project.tasks.withType<KotlinCompile> {
-      //            TODO (#785): revert this setting
-      //            kotlinOptions.allWarningsAsErrors = true
+      kotlinOptions.allWarningsAsErrors = true
```
Comment (PawelLipski) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/BuildUtils.kt:19]: Huh I think this should really be done under #1004, not #785

Comment (PawelLipski) [reply buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/BuildUtils.kt:19]: Whoops no, good, #785, leave as is

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -10,6 +10,15 @@ plugins {
   `kotlin-dsl`
 }
 
+afterEvaluate {
```
Comment (PawelLipski) [inline buildSrc/build.gradle.kts:13]: Hmmm (why?) is this needed? 🤔

File: buildSrc/build.gradle.kts [hunk 4]
```diff
@@ -10,6 +10,15 @@ plugins {
...
+  tasks.withType<KotlinCompile>().configureEach {
+    kotlinOptions {
+      apiVersion = "1.6"
```
Comment (PawelLipski) [inline buildSrc/build.gradle.kts:16]: Also, even if needed, I'd avoid hardcoding the version like this

File: buildSrc/build.gradle.kts (hunk 3 above)
Comment (mkondratek) [reply buildSrc/build.gradle.kts:13]: https://handstandsam.com/2022/04/13/using-the-kotlin-dsl-gradle-plugin-forces-kotlin-1-4-compatibility/

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -39,8 +39,8 @@ pluginPackages-jetbrains-grammarkit = "org.jetbrains.intellij.plugins:gradle-gra
 pluginPackages-jetbrains-intellij = "org.jetbrains.intellij.plugins:gradle-intellij-plugin:1.7.0"
 # @pin because as per https://plugins.jetbrains.com/docs/intellij/kotlin.html#kotlin-standard-library:
 #      "If a plugin supports multiple platform versions, it must (...) target the lowest bundled stdlib version"
-#      which is different in our case (as this is the Kotlin version bundled into IntelliJ 2021.3).
-pluginPackages-jetbrains-kotlin = "org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.10"
+#      which is different in our case (as this is the Kotlin version bundled into IntelliJ 2022.1).
+pluginPackages-jetbrains-kotlin = "org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.20"
```
Comment (PawelLipski) [inline gradle/libs.versions.toml:43]: Pls remove the `@pin` from `pluginPackages-aspectj-postCompileWeaving`

File: buildSrc/build.gradle.kts (hunk 3 above)
Comment (PawelLipski) [reply buildSrc/build.gradle.kts:13]: Ok pls include this comment in code

File: gradle/libs.versions.toml (hunk 5 above)
Comment (mkondratek) [reply gradle/libs.versions.toml:43]: pin must stay; comment updated

File: CHANGE-NOTES.html [hunk 6]
```diff
@@ -1 +1 @@
-<p>Dropped support for IntelliJ 2020.3, 2021.1 and 2021.2.</p>
+<p>Dropped support for IntelliJ 2020.3, 2021.1, 2021.2 and  2021.3.</p>
```
Comment (PawelLipski) [inline CHANGE-NOTES.html:1]: Pls add some _reassuring_ note that the already-released versions of the plugin will be available indefinitely on older IntelliJ versions

Comment (PawelLipski) [reply CHANGE-NOTES.html:1]: :care:

**Review by PawelLipski — APPROVED** [review]
Approving, subject to adding a reassurance into change notes... _psychological aspects of releasology_

**Pull request merged**